pkg/controller/node/node_controller.go

```diff
@@ -1002,6 +1006,27 @@ func (ctrl *Controller) updateCandidateMachines(pool *mcfgv1.MachineConfigPool,
 	return nil
 }
 
+// sortNodeList sorts the list of candidate nodes by label topology.kubernetes.io/zone
+// nodes without label are at end of list and sorted by age (oldest to youngest)
+func sortNodeList(nodes []*corev1.Node) []*corev1.Node {
+	sort.Slice(nodes, func(i, j int) bool {
+		iZone, iOk := nodes[i].Labels["topology.kubernetes.io/zone"]
+		jZone, jOk := nodes[j].Labels["topology.kubernetes.io/zone"]
+		// if both nodes have zone label, sort by zone, push nodes without label to end of list
+		if iOk && jOk {
+			return iZone < jZone
```

Comment:
I think we need to handle the case where the zones are equal here, and compare by timestamp.

Reply:
done via https://github.com/openshift/machine-config-operator/pull/3125